pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -176,26 +188,98 @@ func (c *Reconciler) reconcilePA(ctx context.Context, rev *v1.Revision) error {
 		return fmt.Errorf("revision: %q does not own PodAutoscaler: %q", rev.Name, paName)
 	}
 
+	logger.Debugf("Observed PA Status=%#v", pa.Status)
+	rev.Status.PropagateAutoscalerStatus(&pa.Status)
+
 	// Perhaps tha PA spec changed underneath ourselves?
 	// We no longer require immutability, so need to reconcile PA each time.
 	tmpl := resources.MakePA(rev, deployment)
 	logger.Debugf("Desired PASpec: %#v", tmpl.Spec)
-	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) {
-		diff, _ := kmp.SafeDiff(tmpl.Spec, pa.Spec) // Can't realistically fail on PASpec.
-		logger.Infof("PA %s needs reconciliation, diff(-want,+got):\n%s", pa.Name, diff)
-
+	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) || !annotationsNeedReconcilingForKPA(pa.Annotations, tmpl.Annotations) {
 		want := pa.DeepCopy()
 		want.Spec = tmpl.Spec
-		if pa, err = c.client.AutoscalingV1alpha1().PodAutoscalers(ns).Update(ctx, want, metav1.UpdateOptions{}); err != nil {
-			return fmt.Errorf("failed to update PA %q: %w", paName, err)
+
+		syncAnnotationsForKPA(want.Annotations, tmpl.Annotations)
+
+		// Can't realistically fail on PASpec.
+		if diff, _ := kmp.SafeDiff(want.Spec, pa.Spec); diff != "" {
+			logger.Infof("PA %q spec need reconciliation, diff(-want,+got):\n%s", pa.Name, diff)
+		}
+
+		if diff, _ := kmp.SafeDiff(want.Annotations, pa.Annotations); diff != "" {
+			logger.Infof("PA %q annotations need reconciliation, diff(-want,+got):\n%s", pa.Name, diff)
+		}
+
+		_, err := c.client.AutoscalingV1alpha1().PodAutoscalers(ns).Update(ctx, want, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to update PA %q: %w", want.Name, err)
 		}
 	}
 
-	logger.Debugf("Observed PA Status=%#v", pa.Status)
-	rev.Status.PropagateAutoscalerStatus(&pa.Status)
 	return nil
 }
 
+// syncAnnotationsForKPA will properly handle
+// autoscaling annotations semantics
+//
+// dst should be non-nil
+func syncAnnotationsForKPA(dst, src map[string]string) {
+	// Delete autoscaling annotations from destination map
+	// This ensures that setting these annotation on the Revision is the source of truth
+	for k := range dst {
+		if defaultKPAAnnotations.Has(k) {
+			// Exclude defaulted annotation
+			continue
+		}
+		if strings.HasPrefix(k, autoscaling.GroupName) {
+			delete(dst, k)
+		}
+	}
+
+	// copy source annotations over while preserving existing ones
+	maps.Copy(dst, src)
+}
+
+// this function changes to ensure all annotations in
+// src are present in dst.
+//
+// It will detect deletions for kpa annotations
+func annotationsNeedReconcilingForKPA(dst, src map[string]string) bool {
+	// Check for extra autoscaling annotations that don't exist src
+	for k := range dst {
+		if !strings.HasPrefix(k, autoscaling.GroupName) {
+			continue
+		}
+		// Exclude defaulted annotation
+		if defaultKPAAnnotations.Has(k) {
+			continue
+		}
+
+		if _, ok := src[k]; !ok {
+			// Scaling annotation is in dst but not src
+			// return false to trigger reconciliation
+			return false
```

Comment:
the name of the func implies true should trigger reconcile, but this here says false should trigger reconcile, name if the func is a little confusing